### PR TITLE
Added logger and refactored code

### PR DESCRIPTION
**Pull request by Devas**
Added logger for the crawler;
Removed printing out a stacktrace from the catch block;
Added tests for BookAddingCallback;
Added tests for CustomCallback;
Fixed Code smells as shown in SonarCloud;
Added converter for boogle book and tests for it;
Removed code smell as requested by sonar;

### CODE REVIEW

File: crawler/src/main/java/com/javaacademy/crawler/App.java [hunk 1]
```diff
@@ -13,19 +17,20 @@
...
         googleScrapper.runScrapping();
 
         while (!googleScrapper.areAllCallbacksDone()) {
             try {
                 Thread.sleep(6000);
             } catch (InterruptedException e) {
-                e.printStackTrace();
-                Thread.currentThread().interrupt();
+                AppLogger.logger.log(Level.WARNING,"Exception while waiting", e);
+                        Thread.currentThread().interrupt();
```
Comment (patkkka) [inline crawler/src/main/java/com/javaacademy/crawler/App.java]: ... :)

File: crawler/src/test/java/com/javaacademy/crawler/common/BookAddingCallbackTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Test
+    public void testGetters() {
+        assertEquals(callback.getBooks(), books);
+        assertEquals(callback.getBookstoreName(), bookstoreName);
+    }
+
+    @Test
+    public void testOnResponseSuccess() {
+        BooksWrapper booksWrapper = mock(BooksWrapper.class);
```
Comment (patkkka) [inline crawler/src/test/java/com/javaacademy/crawler/common/BookAddingCallbackTest.java]: name test methods without "test..."

File: crawler/src/test/java/com/javaacademy/crawler/googlebooks/model/GoogleBookConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        assertEquals(bookModel.getIndustryIdentifiers(), convertedIsbns);
+        assertEquals(bookModel.getCategories(), categories);
+        assertEquals(bookModel.getSmallThumbnail(), smallThumbNail);
+        assertEquals(bookModel.getCanonicalVolumeLink(), canonicalVolumeLink);
+        assertEquals(bookModel.getSaleability(), saleability);
+        assertEquals(bookModel.getListPriceAmount(), listPriceAmount);
+        assertEquals(bookModel.getListPriceCurrencyCode(), listPriceCurrencyCode);
+        assertEquals(bookModel.getRetailPriceAmount(), retailPriceAmount);
+        assertEquals(bookModel.getRetailPriceCurrencyCode(), retailPriceCurrencyCode);
+     }
```
Comment (patkkka) [inline crawler/src/test/java/com/javaacademy/crawler/googlebooks/model/GoogleBookConverterTest.java]: ✝️

File: crawler/src/test/java/com/javaacademy/crawler/common/CustomCallbackTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    @Test
+    public void testEqualsAndGetter() {
+        CustomCallback<BooksWrapper> callback2 = new CustomCallback<>(consumer);
+        assertTrue(callback2.equals(customCallback));
+        Consumer consumer2 = mock(Consumer.class);
+        callback2 = new CustomCallback<>(consumer2);
+        assertEquals(callback2.getConsumer(), consumer2);
+        assertFalse(customCallback.equals(callback2));
+    }
+
```
Comment (kukulastudent) [inline crawler/src/test/java/com/javaacademy/crawler/common/CustomCallbackTest.java]: no gwt

File: springapp/src/test/java/com/javaacademy/robot/service/BookServiceTest.java [hunk 5]
```diff
@@ -24,14 +24,14 @@ public void testAutowiring() {
...
-    public void testSave() throws Exception {
+    public void testSave() {
         Book book = new Book(1L, "DummyName");
         boolean returnValue = bookService.saveBook(book);
         assertTrue(returnValue);
     }
 
     @Test
-    public void testGetAllBooks() throws Exception {
+    public void testGetAllBooks() {
```
Comment (kukulastudent) [inline springapp/src/test/java/com/javaacademy/robot/service/BookServiceTest.java]: tests in method

**Pull request merged**